Feat: Implement multi-day calendar header and fix date visibility

Aaratrika-Shelly opened the pull request:
This PR resolves the issue regarding the calendar date visibility. After investigation, it was determined that the multi-day header needed to be implemented as a new feature.

### Changes Made

- **Implemented Multi-Day Header:** The calendar now displays a header for each visible day, perfectly aligned with its corresponding column in the grid.
- **Fixed Visibility Bug:** The text for all dates is now correctly styled with dark mode variants, ensuring they are visible in both light and dark themes.
- **Improved UI Structure:** Refactored the component's JSX to logically separate the date header from the top navigation, resulting in a cleaner and more maintainable layout.

This PR is based on the `test-dash` branch as requested and delivers the feature shown in the issue's screenshot.

### Final Result

<img width="1914" height="966" alt="Screenshot 2025-10-22 181909" src="https://github.com/user-attachments/assets/0f947f48-afe5-4179-b8c5-375db3dc5276" />


Closes #31

PriyanshuShekhar10 commented:
<img width="1919" height="900" alt="image" src="https://github.com/user-attachments/assets/d7d076a2-733d-4b30-87fd-94d38587e154" />
@Aaratrika-Shelly This is alright but can you make the date fixed at the top so even when the user scrolls, it shows the date?

also pls rebase your branch

Aaratrika-Shelly commented:
@PriyanshuShekhar10 
Done! I've rebased the branch and implemented the sticky header as you requested. It should be ready for a final review. Thanks!

kanishk6103 commented:
> @PriyanshuShekhar10 Done! I've rebased the branch and implemented the sticky header as you requested. It should be ready for a final review. Thanks!

<img width="1391" height="652" alt="image" src="https://github.com/user-attachments/assets/5371b5fb-4b26-4444-b249-520282a167e1" />

There is an offset visible here, the preview box seems to be misplaced with respect to the cursor, can you fix that? @Aaratrika-Shelly

Aaratrika-Shelly commented:
Hi @kanishk6103 and @PriyanshuShekhar10,
Just wanted to gently follow up on this PR. I've completed all the requested changes (sticky header and hover offset fix), and the branch has been rebased.
With the Hacktoberfest deadline getting closer, I would be very grateful if you could take a moment for a final review when you have a chance.
Please let me know if there's anything else you need from my end. Thank you again for your guidance on this feature!"

kanishk6103 commented:
> Hi @kanishk6103 and @PriyanshuShekhar10, Just wanted to gently follow up on this PR. I've completed all the requested changes (sticky header and hover offset fix), and the branch has been rebased. With the Hacktoberfest deadline getting closer, I would be very grateful if you could take a moment for a final review when you have a chance. Please let me know if there's anything else you need from my end. Thank you again for your guidance on this feature!"

Hey we checked the deployment, please test it properly, session booking does not work as it should, the booking time and booked time is inconsistent which shouldn't be the case. Also there seems to be no reason to add mongoose, is there any particular reason you have added it?

<img width="1384" height="740" alt="image" src="https://github.com/user-attachments/assets/f359e53d-a204-4e60-9599-2741e8162514" />

Aaratrika-Shelly committed:
Feat: Implement multi-day calendar header and fix date visibility

Aaratrika-Shelly commented:
"Hi @kanishk6103 @PriyanshuShekhar10 ,
I've pushed the final updates. I have addressed all the feedback:
The branch has been cleaned and rebased.
The multi-day header is implemented and is sticky.
I've adjusted the coordinate calculations, which fixes both the hover preview offset and the booking time inconsistency.
The feature should now be pixel-perfect and fully functional. Ready for a final review. Thanks for your guidance!"

Aaratrika-Shelly commented:
hi @PriyanshuShekhar10 @kanishk6103 .
are there any changes required before the final merge?
